torchsynth/module.py

```diff
@@ -49,28 +58,38 @@ def __init__(self, synthglobals: SynthGlobals, **kwargs: Dict[str, T]):
         """
         nn.Module.__init__(self)
         self.synthglobals = synthglobals
+        self.device = device
+        self.synthglobals.to(device)
         self.torchparameters: nn.ParameterDict = nn.ParameterDict()
         # If this module needs a random seed, here it is
         self.seed: Optional[int] = None
 
         if self.parameter_ranges:
+            # TODO: Is this copying thing what we want?
+            # We want to create copies of the parameter ranges otherwise each instance
+            # of the same module type (ex. ADSR) will reference the same param range
+            parameter_ranges = copy.deepcopy(self.parameter_ranges)
```

Comment:
@turian -- not sure how you had the optuna hyperparam tuning going so you may have already thought about this, but basically every instance of an ADSR shared the same hyperparams, so if we updated the attack curve of one of the adsrs, then that would be the curve for all of them. If we want to adjust them all independently then we want this deepcopy approach here so each parameter has their own hyperparameters. Alternatively we decide that each instances of an ADSR should have the same hyperparams, in which case we maybe update our API a bit. However, now that I'm thinking about this maybe you had it setup correctly to begin with and I mucked it up with my hyperparam changes ... :/

Reply:
hmmmm I wonder where this bug entered. How did you see it? Also, does this fix completely prevent anyone from footgunning themselves with this in the future?

My current optuna runs which are on main (2b416f98ab6756b672f2d9ec4cbc33c387540b97) don't appear to have tied ADSR params:

![image](https://user-images.githubusercontent.com/65918/112719254-6146de00-8ef8-11eb-9333-0ba4fb627d35.png)

![image](https://user-images.githubusercontent.com/65918/112719265-6c017300-8ef8-11eb-9e3b-85742bd79e03.png)

Like those look fine to me?

Reply:
so somehow this came thru main, or thru my PR, or thru yours. I think we should understand it a little more deeply just to make sure it doesn't shoot us in the foot.

Reply:
Okay shit this bug is in main.

Reply:
But your fix appears to work

Reply:
What I don't love about this fix tho is that if you're in a module and you do self.parameter_ranges, you get the wrong thing.

What about either:

1) `self.parameter_ranges = copy.deepcopy(self.parameter_ranges)`
with a comment explaining that otherwise they are tied across module instances. Seems clean and works?

2) Rename it to orig_parameter_ranges in each module so people don't footgun themselves with it.

Reply:
Yes, good point. Unfortunately 1. doesn't work. I tried that approach. 2. is a good idea and would work. I'm into that.

As another point it would be great to be able to quickly save and load hyperparam settings. What if we had a json format for hyperparams for a particular synth? Say, Voice could have a default hyperparam setting for Synth1B1 that gets loaded automatically. But others could also be loaded as well.

Reply:
Issue for saving and loading hyperparam settings: #232